src/components/DayPickerSingleDateController.jsx

```diff
@@ -548,39 +549,44 @@ export default class DayPickerSingleDateController extends React.Component {
       isFocused,
       isRTL,
       phrases,
+      onOutsideClick,
     } = this.props;
 
     const { currentMonth, visibleDays } = this.state;
 
     return (
-      <DayPicker
-        orientation={orientation}
-        enableOutsideDays={enableOutsideDays}
-        modifiers={visibleDays}
-        numberOfMonths={numberOfMonths}
-        onDayClick={this.onDayClick}
-        onDayMouseEnter={this.onDayMouseEnter}
-        onDayMouseLeave={this.onDayMouseLeave}
-        onPrevMonthClick={this.onPrevMonthClick}
-        onNextMonthClick={this.onNextMonthClick}
-        monthFormat={monthFormat}
-        withPortal={withPortal}
-        hidden={!focused}
-        hideKeyboardShortcutsPanel={hideKeyboardShortcutsPanel}
-        initialVisibleMonth={() => currentMonth}
-        firstDayOfWeek={firstDayOfWeek}
-        navPrev={navPrev}
-        navNext={navNext}
-        renderMonth={renderMonth}
-        renderDay={renderDay}
-        renderCalendarInfo={renderCalendarInfo}
-        isFocused={isFocused}
-        getFirstFocusableDay={this.getFirstFocusableDay}
-        onBlur={this.onDayPickerBlur}
-        phrases={phrases}
-        daySize={daySize}
-        isRTL={isRTL}
-      />
+      <OutsideClickHandler
+        onOutsideClick={onOutsideClick}
```

Comment:
Yes, that way if people don't want outside click behavior, they wouldn't incur any overhead.